### PR TITLE
[3.0] Fix tests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Should make tests green again on travis and appveyor

### CODE REVIEW

File: src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php [hunk 1]
```diff
@@ -355,13 +351,9 @@ public function testFormatWithConstructorTimezone()
 
     public function testFormatWithDateTimeZone()
     {
-        if (defined('HHVM_VERSION_ID')) {
-            $this->markTestSkipped('This test cannot work on HHVM. See https://github.com/facebook/hhvm/issues/5875 for the issue.');
-        }
-
-        $formatter = $this->getDateFormatter('en', IntlDateFormatter::MEDIUM, IntlDateFormatter::SHORT, new \DateTimeZone('GMT+03:00'), IntlDateFormatter::GREGORIAN, 'zzzz');
+        $formatter = $this->getDateFormatter('en', IntlDateFormatter::MEDIUM, IntlDateFormatter::SHORT, new \DateTimeZone('GMT'), IntlDateFormatter::GREGORIAN, 'zzzz');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php]: ping @stof was there a specific reason to test with 'GMT+03:00' here? Because it doesn't work on 5.5.9 (only on 5.5.10+)

Comment (stof) [inline src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php]: well, the goal was to ensure that this syntax was supported when converting DateTimeZone to Intl timezone ids (there is a specific handling for them).

We might add another test using GMT only and keep the test with ``GMT+03:00`` but skipping it in versions not supporting it.

Btw, this should be done in 2.3+ if we go this way

File: src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php [hunk 2]
```diff
@@ -47,11 +46,6 @@ class UniqueEntityValidatorTest extends AbstractConstraintValidatorTest
      */
     protected $repository;
 
-    protected function getApiVersion()
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php]: shoudln't this be kept for lowest deps (which would use the 2.x validator component) ?

File: src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php]: hmm, given the test failure, it looks like we have an issue with the support of ``GMT`` in the stub code

File: src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php]: I reverted the change, it's not related to this pr

File: src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Tests/DateFormatter/AbstractIntlDateFormatterTest.php]: I reverted the change and skipped the test on 5.5.9. We should open a PR on 2.3 now.

File: src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php (hunk 2 above)
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php]: it would be great to send a separate PR with this cleanup as you have it though

File: src/Symfony/Component/Validator/Tests/Constraints/NotIdenticalToValidatorTest.php [hunk 3]
```diff
@@ -87,9 +87,6 @@ public function provideInvalidComparisons()
             array($object, '2', $object, '2', __NAMESPACE__.'\ComparisonTest_Class'),
         );
 
-        $immutableDate = new \DateTimeImmutable('2000-01-01');
-        $comparisons[] = array($immutableDate, 'Jan 1, 2000, 12:00 AM', $immutableDate, 'Jan 1, 2000, 12:00 AM', 'DateTime');
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/NotIdenticalToValidatorTest.php]: why is it removed ?

Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Tests/Constraints/NotIdenticalToValidatorTest.php]: because it's wrong, and already tested thanks to https://github.com/nicolas-grekas/symfony/blob/m-fx/src/Symfony/Component/Validator/Tests/Constraints/AbstractComparisonValidatorTestCase.php#L37

Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/NotIdenticalToValidatorTest.php]: ah, I was not aware of this nice trick

**Comment by nicolas-grekas** [comment]
PR is ready to merge. I can't reproduce the failures on the doctrine bridge with a fresh php7.
If someone can, please tell me where is the issue, otherwise the php7 on travis is broken and may be fixed tomorrow.

**Comment by nicolas-grekas** [comment]
Upgrading to doctrine/orm ^2.4.6 as lowest version fixes the remaining test failures.

File: composer.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
     ],
     "require": {
         "php": ">=5.5.9",
-        "doctrine/common": "~2.3",
+        "doctrine/common": "~2.4",
```
Comment (Tobion) [inline composer.json]: if these changes fix tests for php 7, should they not be done on symfony 2.3 or 2.7/2.8 instead? I assume they would fail there as well since 3.0 doesn't have anything new yet, afaik.

Comment (nicolas-grekas) [inline composer.json]: I agree

Comment (stof) [inline composer.json]: I'm totally fine with bumping the Doctrine base versions btw. 2.2 and 2.3 are long EOLed.

File: src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php [hunk 5]
```diff
@@ -855,6 +855,10 @@ private function getInt64Value($value)
             return false;
         }
 
+        if (PHP_INT_SIZE !== 8 && ($value > self::$int32Range['positive'] || $value < self::$int32Range['negative'])) {
+            return (float) $value;
```
Comment (stof) [inline src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php]: the phpdoc needs to be updated though, as it says it returns ``int|false``

Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php]: See #15672

**Comment by nicolas-grekas** [comment]
Reverted the doctrine upgrade: that was a temporary bug in php7

File: src/Symfony/Component/Validator/Tests/Constraints/NotIdenticalToValidatorTest.php (hunk 3 above)
Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/NotIdenticalToValidatorTest.php]: shouldn't this be done in older branches though ?
